### ISSUE TITLE
Settings Framework doesn't support mod settings (non mission/server)

### ISSUE THREAD

**Issue by Grezvany13** [CBATeam/CBA_A3#504]
**Arma 3 Version:** `1.62` (stable)
**CBA Version:** `3.6.2.0` (stable)

**Mods:**
- `@CBA_A3`
- `@CustomMod`

**Description:**
The Settings Framework is a great tool to add custom settings for a mod inside a mission, however it seems that it currently doesn't support local settings for a mod/addon without being tied to a mission (like a userconfig file would).

**Steps to reproduce:**
1) Start Arma3 with `@CBA` and `@CustomMod` (which uses `cba_settings_fnc_init` to add settings)
2) Go to Options -> Game -> Configure Addons
3) _You'll see that the Addons select box is empty_

**Where did the issue occur?**
- Singleplayer
- Editor

**RPT log file:**
- N/A

This is either a bug, a (missing) feature or simply me not understanding how to implement it. However reading the scripts of the Settings UI, it doesn't seem to support `profileNamespace`.
This feature would be very useful for mod developers to include mod specific settings which may be set client side (and could be forcefully overwritten by mission/server settings).

**Comment by commy2** [comment]
Settings Client/Server values are stored in the `profileNamespace`
https://github.com/CBATeam/CBA_A3/blob/master/addons/settings/fnc_set.sqf#L61

Settings Mission values are stored in the `mission.sqm`
https://github.com/CBATeam/CBA_A3/blob/master/addons/settings/fnc_set.sqf#L96

The setting should show up just fine. Can you upload a screenshot? Are you testing in 3den? The thing with 3den is, that it's bugged with the `preInit = 1;` CfgFunctions and doesn't execute postInit at all unless you preview the mission. To make sure the setting shows up there, you should ideally use CBA XEH preInit (which get's around this bug with some band aid magic).

**Comment by Grezvany13** [comment]
Thanks @commy2, I figured out what went wrong.

Since I had some issues with calling the scripts (and therefor didn't work), I messed up the event handlers during debugging which made it worse.

And I found out, after reading some issues at ACE3, that it's required to pre-load a world map when starting the game, otherwise PreInit doesn't work (and the "Configure Addons" is grayed out).


For people who try to make it work as well and end up at this issue I've created a simple gist with some example code: [ArmA 3 - @CBA_A3 - CBA Settings Framework](https://gist.github.com/Grezvany13/fefae80ecb17e3cf826f5520c4b1d732)
